projects/core/components/button/button-options.ts

```diff
@@ -0,0 +1,27 @@
+import {InjectionToken} from '@angular/core';
+import {TuiAppearance} from '@taiga-ui/core/enums';
+import {TuiSizeXL, TuiSizeXS} from '@taiga-ui/core/types';
+import {PolymorpheusContent} from '@tinkoff/ng-polymorpheus';
+
+export interface ButtonOptions {
+    readonly size: TuiSizeXS | TuiSizeXL;
+    readonly appearance: TuiAppearance | string;
+    readonly shape: 'square' | 'rounded' | null;
+    readonly icon: PolymorpheusContent;
```

Comment:
I don't think it makes sense to configure icon with DI. It is different for each instance anyway.

Reply:
removed 👍